src/txn/txn.c

```diff
@@ -946,9 +946,10 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
          * Scan the history store for the given btree and key with maximum start timestamp to let
          * the search point to the last version of the key.
          */
-        WT_ERR_NOTFOUND_OK(__wt_hs_cursor_position(
-                             session, hs_cursor, hs_btree_id, &op->u.op_row.key, WT_TS_MAX, NULL),
-          true);
+        WT_WITH_TXN_ISOLATION(
+          session, WT_ISO_READ_UNCOMMITTED, ret = __wt_hs_cursor_position(session, hs_cursor,
+                                              hs_btree_id, &op->u.op_row.key, WT_TS_MAX, NULL));
```

Comment:
It's bad that the wrapping of history store operations is done outside of the history store code.

Can you instead do this:
```
int __wt_hs_cursor_position(...) {
    WT_WITH_TXN_ISOLATION(session, WT_ISO_READ_UNCOMMITTED, ret = __wt_hs_cursor_position_int(...));
    return (ret);
}
```

And rename the existing `__wt_hs_cursor_position` to be called `__wt_hs_cursor_position_int`?

Similarly adding a wrapper for history store `WT_CURSOR::search_near()`, `WT_CURSOR::next()` and `WT_CURSOR::prev()`. So that there are four wrapper functions that set the isolation level, and all cursor operations on the history store go through them.